### PR TITLE
Basic Narg Functionality

### PR DESCRIPTION
**Pull request by alanxoc3**
I needed Narg functionality in my application, and I was able to achieve what I needed with a few minor, backwards compatible changes to the library. This doesn't have to be merged, but it might help bring some good ideas to the discussion (I noticed #20, #22, and #38).

Also, my application doesn't use sub commands, so that may be why this change works well for me.

### CODE REVIEW

File: argument.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type Arg interface {
 	GetOpts() *Options
 	GetSname() string
 	GetLname() string
+	IsFlag() bool
```
Comment (alanxoc3) [inline argument.go:30]: This IsFlag line is unrelated to the topic of the pull request, but it was useful for me to get a nice looking help menu. Example:
```
Options:
  -h  --help      Print help information.
  -V  --version   Concards build information..
  -r  --review    Show cards available to be reviewed.
  -m  --memorize  Show cards available to be memorized.
  -d  --done      Show cards not available to be reviewed or memorized.
  -p  --print     Prints all cards, one line per card.
  -n  --number n  How many cards to review.
  -E  --editor E  Which editor to use. Defaults to "$EDITOR".
  -M  --meta M    Path to meta file. Defaults to "$CONCARDS_META" or "~/.concards-meta".
```

Note that in my custom help menu function, if the argument is a flag, I'm not printing an argument name. And conversely, if the argument is not a flag, there is an argument name (same as the short name).

Comment (akamensky) [reply argument.go:30]: That should be its own PR please. 1 PR == 1 functionality change.

Also that is not really common way for building help message. Arguably there are no standards of doing this per se. But there are recognized common formats for help messages and that wouldn't be it.

File: argument.go [hunk 2]
```diff
@@ -41,6 +42,10 @@ func (o arg) GetLname() string {
 	return o.lname
 }
 
+func (o arg) IsFlag() bool {
+	return o.size == 1
```
Comment (akamensky) [inline argument.go:46]: `o.size` does not mean it is a flag. flag would be a type of the interface successfully cast to boolean.

**Comment by akamensky** [comment]
I am confused here, the top post mentions 3 issues. 2 of them talk about positional arguments, another about nargs. I should probably note -- positional arguments != nargs.

This change is missing any test cases (therefore failure of code coverage checks).

**Pull request closed without merging**